tables/tabulate-molecular-alterations.R

```diff
@@ -0,0 +1,433 @@
+  dplyr::distinct()
+
+  
+# At this point, alterations_presence_df is a Y/N tibble, but it contains only `sample_id`, not DNA and RNA biospecimen ids.
+# The final step is to add that information in, but we need to handle different groups separately:
+# - First, the fully non-ambiguous samples
+# - Second, the ambiguous samples: where DNA and RNA ids cannot be reliably linked, OR 
+#   where there are multiple same-modality experiments (since we are working with all samples, not `primary-only`)
+
+## Combine alterations_presence_df with biospecimen sample ids ----------------------
+  
+# First, let's "combine" the ambiguous ids: 
+# For multiple DNA or RNA for a given sample_id, convert to a semi-colon separated single value
+# For example, two DNA ids on separate rows named BS_1 and BS_2 would be collapsed into `BS_1; BS2`
+merged_ambiguous_ids <- tumor_sample_ids_df %>%
+  # get only samples of interest
+  dplyr::filter(ambiguous | multiple_exp_strategy) %>%
+  # for each sample_id, combine biospecimen ids by modality to have 
+  # at most one "value" (semi-colon separated list) for each modality 
+  dplyr::group_by(sample_id, modality) %>%
+  dplyr::summarize(Kids_First_Biospecimen_ID = paste(Kids_First_Biospecimen_ID, collapse="; ")) %>%
+  dplyr::ungroup() %>%
+  # bring back the other variables from `tumr_sample_ids_df`
+  dplyr::inner_join(
+    dplyr::select(
+      tumor_sample_ids_df, 
+      sample_id, 
+      broad_histology_display,
+      cancer_group_display,
+      germline_sex_estimate
+    )
+  ) %>%
+  dplyr::distinct() %>%
+  # add indicator that these are ambiguous mappings
+  dplyr::mutate(ambiguous_biospecimen_mapping = TRUE)
+
+
+# Now, subset to just the identifiable ids and bind_rows with `merged_ambiguous_ids` for final processing
+prepared_ids_df <- tumor_sample_ids_df %>%
+  # only ids of interest
+  dplyr::filter(!(ambiguous), 
+                !(multiple_exp_strategy)) %>%
+  dplyr::select(-ambiguous, -multiple_exp_strategy) %>%
+  # add indicator that these are NOT ambiguous mappings
+  dplyr::mutate(ambiguous_biospecimen_mapping = FALSE) %>%
+  # combine with prepared ambiguous samples
+  dplyr::bind_rows(merged_ambiguous_ids) %>%
+  dplyr::distinct()
+
+# Again, check that we've got everything: there should be `openpbta_n_tumors` 
+#  unique sample_id values
+if (!(length(unique(tumor_sample_ids_df$sample_id))) == openpbta_n_tumors) {
+  stop("An error occurred while dealing with ambiguous vs. identifiable biospecimen ids.")
+} 
+
+## Finally, process the full data for export:
+zenodo_df <- prepared_ids_df %>%
+  tidyr::spread(modality, Kids_First_Biospecimen_ID) %>%
+  # join with alterations themselves
+  dplyr::inner_join(
+    alterations_presence_df,
+    by = "sample_id"
+  ) %>%
+  # update columns and their order
+  dplyr::select(
+    sample_id, 
+    Kids_First_Biospecimen_ID_DNA = DNA,
+    Kids_First_Biospecimen_ID_RNA = RNA,
+    ambiguous_biospecimen_mapping,
```

Comment:
There may be two values here, right? One each for DNA & RNA? do we need to account for that in the spread?